src/lib/util/state.ts

```diff
@@ -35,9 +35,11 @@ export const loadState = (data: string): void => {
 		const stateStr = fromBase64(data);
 		console.log(`Tring to load state: ${stateStr}`);
 		state = JSON.parse(stateStr);
-		if (typeof state.mermaid !== 'string') {
-			state.mermaid = JSON.stringify(state.mermaid, null, 2);
-		}
+		const mermaidConfig =
+			typeof state.mermaid === 'string' ? JSON.parse(state.mermaid) : state.mermaid;
+		delete mermaidConfig.securityLevel; // Prevent setting overriding securityLevel when loading state to mitigate possible XSS attack
```

Comment:
```suggestion
              if(mermaidConfig.securityLevel) {
                  alert(`securityLevel was removed from config. Please add "securityLevel":"${mermaidConfig.securityLevel}" to your config if you trust the source of this Diagram`);
		    delete mermaidConfig.securityLevel; // Prevent setting overriding securityLevel when loading state to mitigate possible XSS attack
		}
```

Reply:
One issue with this is the `/view` endpoint will never work.
We could use the `confirm` prompt to give the user an option to retain the config.

```js
confirm(`Removing "securityLevel":"${mermaidConfig.securityLevel}" from the config for safety.\nClick Cancel if you trust the source of this Diagram.`);
```

Reply:
LGTM but in this case I think it's appropriate to check the value of securityLevel before prompting the user. Because there is no need to bother him if the securityLevel is set to "strict" anyway.